Omitted left | right value of text-emphasis-position defaults to right

fantasai opened the pull request:
#### 29304d7c911953125d52935c1281063b7157d707
<pre>
Omitted left | right value of text-emphasis-position defaults to right
<a href="https://bugs.webkit.org/show_bug.cgi?id=279217">https://bugs.webkit.org/show_bug.cgi?id=279217</a>
<a href="https://rdar.apple.com/problem/135366186">rdar://problem/135366186</a>

Reviewed by Alan Baradlay.

If [ left | right ] is omitted from text-emphasis-position, it defaults to right.
<a href="https://www.w3.org/TR/css-text-decor/#text-emphasis-position-property">https://www.w3.org/TR/css-text-decor/#text-emphasis-position-property</a>

* LayoutTests/fast/repaint/text-emphasis-h.html:
* LayoutTests/fast/text/emphasis-vertical-over-right-expected.html:
* LayoutTests/fast/text/emphasis-vertical-over-right.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/text-emphasis-position-default-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/text-emphasis-position-default-001.html: Added.
* Source/WebCore/layout/formattingContexts/inline/InlineFormattingUtils.cpp:
(WebCore::Layout::InlineFormattingUtils::textEmphasisForInlineBox):
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::emphasisMarkExistsAndIsAbove):

Canonical link: <a href="https://commits.webkit.org/283275@main">https://commits.webkit.org/283275@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a4441fb8c8dc448df63eb63596d9c737fbce55b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65759 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/45132 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18378 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69785 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16368 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/67877 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52931 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16650 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52786 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11367 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68826 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41668 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56914 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33414 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38343 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14294 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/15244 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/60196 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14636 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71491 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9714 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/14069 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/60108 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9746 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56980 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60385 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14499 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/8019 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1670 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40940 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/42016 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/43199 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41760 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderText.cpp:
```diff
@@ -2097,24 +2097,9 @@ std::optional<bool> RenderText::emphasisMarkExistsAndIsAbove(const RenderText& r
+        isAbove = !emphasisPosition.contains(TextEmphasisPosition::Left);
 
     auto findRubyAnnotation = [&]() -> RenderBlockFlow* {
         for (auto* baseCandidate = renderer.parent(); baseCandidate; baseCandidate = baseCandidate->parent()) {
```
I am gonna get rid of this function. there's no reason for having it on RenderText

fantasai replied:
Ok, but let me land this first? :)

fantasai replied:
It comes with tests!

alanbaradlay replied:
oh absolutely! please land!

fantasai committed:
Omitted left | right value of text-emphasis-position defaults to right
https://bugs.webkit.org/show_bug.cgi?id=279217
rdar://problem/135366186

Reviewed by Alan Baradlay.

If [ left | right ] is omitted from text-emphasis-position, it defaults to right.
https://www.w3.org/TR/css-text-decor/#text-emphasis-position-property

* LayoutTests/fast/repaint/text-emphasis-h.html:
* LayoutTests/fast/text/emphasis-vertical-over-right-expected.html:
* LayoutTests/fast/text/emphasis-vertical-over-right.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/text-emphasis-position-default-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-text-decor/text-emphasis-position-default-001.html: Added.
* Source/WebCore/layout/formattingContexts/inline/InlineFormattingUtils.cpp:
(WebCore::Layout::InlineFormattingUtils::textEmphasisForInlineBox):
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::emphasisMarkExistsAndIsAbove):

Canonical link: https://commits.webkit.org/283275@main
